utils/doclint/check_public_api/index.js

```diff
@@ -295,12 +319,15 @@ function compareDocumentations(actual, expected) {
     const actualName = actual.name.replace(/[\? ]/g, '');
     // TypeScript likes to add some spaces
     const expectedName = expected.name.replace(/\ /g, '');
-    if (expectedName !== actualName) {
-      const namingMismatchIsExpected = namingMisMatchInTypeIsExpected(source, actualName, expectedName);
+    const namingMismatchIsExpected = namingMisMatchInTypeIsExpected(source, actualName, expectedName);
+    if (expectedName !== actualName && !namingMismatchIsExpected)
+      errors.push(`${source} ${actualName} != ${expectedName}`);
 
-      if (!namingMismatchIsExpected)
-        errors.push(`${source} ${actualName} != ${expectedName}`);
-    }
+
+    /* If we got a naming mismatch and it was expected, don't check the properties
+     * as they will likely be considered "wrong" by DocLint too
```

Comment:
Nit: Comments should read like sentences:

```suggestion
     * as they will likely be considered "wrong" by DocLint too.
```